engines/router/missionctl/handlers/http_handler.go

```diff
@@ -100,13 +103,17 @@ func (h *httpHandler) getPrediction(
 		if httpErr != nil {
 			return nil, httpErr
 		}
+		// Merge Enricher response headers with original request headers
+		for key, value := range resp.Header() {
+			postEnrichmentResponseHeader.Set(key, value[0])
```

Comment:
Instead, this can be:
```golang
for key := range resp.Header() {
    postEnrichmentResponseHeader.Set(key, resp.Header().Get(key))
}
```

`resp.Header().Get(key)` effectively does `value[0]`. It also handles 0 length value.

Reply:
Alright I'll make that change :)